src/wallet/wallet.cpp

```diff
@@ -5261,8 +5277,11 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
 
         // No need to read and scan block if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        while (pindexRescan && walletInstance->nTimeFirstKey && (pindexRescan->GetBlockTime() < (walletInstance->nTimeFirstKey - TIMESTAMP_WINDOW))) {
-            pindexRescan = chainActive.Next(pindexRescan);
+        // unless a full rescan was requested
+        if (gArgs.GetArg("-rescan", 0) != 2) {
```

Comment:
Say someone puts in `-rescan=3` should this still be true?

Reply:
how about 4a0adc2a6586ded394fba64e9e978a5a1f58e02a?